ADR/0035-artifact-streams.md

```diff
@@ -0,0 +1,150 @@
+# 35. Artifact Streams
+
+Date started: 2024-02-05
+
+## Status
+
+Proposed
+
+Relates to:
+
+<!-- ... a lot ... -->
+
+Supercedes:
+
+<!-- ... a lot ... -->
+
+## Authors
+
+- Andrew M
+
+## Context
+
+Since the beginning of AppStudio, we have had a dependency on the [Component] and [Application] model that
+originated in the Hybrid Application Service. A [Component] is a single containerized artifact that has
+information on how to build and deploy recorded in a linked [Devfile]. Multiple [Components] can be grouped
+together into a single [Application] to indicate that the should be deployed together via GitOps.
+
+Along with the [gitops-service] which was decided to stop deploying in [ADR 32. Decoupling Deployment],
+the controllers' APIs and the UI for AppStudio have been tightly coupled with the concept of [Application]s
+and [Component]s. For example, 
+
+- PipelineRuns triggered by PipelinesAsCode produced a single image that is tied to a specific [Component]
+  which is assumed to be an OCI artifact.
+- The AppStudio UI (HAC) display PipelineRuns for all [Component]s in a single [Application] together.
+- [Component]s are tied to artifacts produced from a single a git repository and branch.
+
+This model presented multiple challenges including:
+
+- If a user wants to build a new branch of an already onboarded git repository, they would need to create
+  a new [Component] (with a workspace-unique name).
```

Comment:
@arewm  what if we don't want previous branch to be built any further? Can be existing component updated to ref a new branch?